crates/bevy_ecs/src/system/system.rs

```diff
@@ -94,6 +94,32 @@ pub trait System: Send + Sync + 'static {
     /// of this system into the world's command buffer.
     fn queue_deferred(&mut self, world: DeferredWorld);
 
+    /// Validates that all systems parameters can be acquired.
+    /// If not, the system should not be ran.
+    ///
+    /// # Safety
+    ///
+    /// - The caller must ensure that `world` has permission to read any world data
+    ///   registered in [`Self::archetype_component_access`]. There must be no conflicting
+    ///   simultaneous accesses while the system is running.
+    /// - The method [`Self::update_archetype_component_access`] must be called at some
+    ///   point before this one, with the same exact [`World`]. If `update_archetype_component_access`
+    ///   panics (or otherwise does not return for any reason), this method must not be called.
+    unsafe fn validate_param_unsafe(&self, world: UnsafeWorldCell) -> bool;
+
+    /// Validates that all systems parameters can be acquired.
+    /// If not, the system should not be ran.
+    /// This is a safe version of [`System::validate_param_unsafe`]
+    /// that runs on exclusive, single-threaded `world` pointer.
+    fn validate_param(&mut self, world: &mut World) -> bool {
```

Comment:
Is there a reason this methods takes a `&mut World` but only uses it in a read-only way?